librz/core/cmd/cmd_analysis.c

```diff
@@ -2918,83 +2872,96 @@ static RzCmdStatus analysis_function_vars_del(RzCore *core, RzAnalysisVarKind ki
 	return RZ_CMD_STATUS_OK;
 }
 
-static RzCmdStatus analysis_function_vars_del_all(RzCore *core, RzAnalysisVarKind kind) {
+static RzCmdStatus analysis_function_vars_del_all(RzCore *core, RzAnalysisVarStorageType kind) {
 	RzAnalysisFunction *fcn = analysis_get_function_in(core->analysis, core->offset);
 	if (!fcn) {
 		return RZ_CMD_STATUS_ERROR;
 	}
-	rz_analysis_function_delete_vars_by_kind(fcn, kind);
+	rz_analysis_function_delete_vars_by_storage_type(fcn, kind);
 	return RZ_CMD_STATUS_OK;
 }
 
-static RzCmdStatus analysis_function_vars_getsetref(RzCore *core, int delta, ut64 addr, RzAnalysisVarKind kind, RzAnalysisVarAccessType access_type) {
+static RzCmdStatus analysis_function_vars_getsetref(RzCore *core, RzAnalysisVarStorage *stor, ut64 addr, RzAnalysisVarAccessType access_type) {
 	RzAnalysisFunction *fcn = analysis_get_function_in(core->analysis, core->offset);
 	if (!fcn) {
 		return RZ_CMD_STATUS_ERROR;
 	}
 
-	RzAnalysisVar *var = rz_analysis_function_get_var(fcn, kind, delta);
+	RzAnalysisVar *var = rz_analysis_function_get_var_at(fcn, stor);
 	if (!var) {
-		RZ_LOG_ERROR("core: Cannot find variable with delta %d\n", delta);
+		switch (stor->type) {
+		case RZ_ANALYSIS_VAR_STORAGE_REG:
+			RZ_LOG_ERROR("core: Cannot find variable for reg %s\n", stor->reg);
+			break;
+		case RZ_ANALYSIS_VAR_STORAGE_STACK:
+			RZ_LOG_ERROR("core: Cannot find variable with stack offset %" PFMT64d "\n", stor->stack_off);
+			break;
+		}
 		return RZ_CMD_STATUS_ERROR;
 	}
 
 	RzAnalysisOp *op = rz_core_analysis_op(core, addr, 0);
 	const char *ireg = op ? op->ireg : NULL;
-	if (kind == RZ_ANALYSIS_VAR_KIND_SPV) {
-		delta -= fcn->maxstack;
+	st64 addend = 0;
+	if (stor->type == RZ_ANALYSIS_VAR_STORAGE_STACK) {
+		// TODO: this is wrong. What we need is not the address on the stack, but
```

Comment:
Probably `FIXME` not `TODO` then

Reply:
Did we ever specify any difference between `FIXME` and `TODO`? I personally always write `TODO`.

Reply:
Not really. Not that important anyway.